pkg/querier/iterators/chunk_merge_iterator.go

```diff
@@ -30,7 +30,8 @@ type chunkMergeIterator struct {
 }
 
 // NewChunkMergeIterator creates a chunkenc.Iterator for a set of chunks.
-func NewChunkMergeIterator(cs []chunk.Chunk, _, _ model.Time) chunkenc.Iterator {
+// TODO: reuse iterators.
```

Comment:
[nit] Not sure this TODO adds much value. If you're going to do it in a follow up PR, then there's no need for this TODO. If you're not going to do it in a follow up PR, then this TODO won't help much to find stuff to work on (an issue may be better).

Reply:
I do not plan to implement iterator re-use for them. I will remove the TODO.